File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_SEGA_LEONEL_ISAIAS/Polimorfisom0/2.java
public class Videojuego {
    private String nombre;
    private String plataforma;
    private int jugadores;

    public Videojuego() {
        this.nombre = "Sin nombre";
        this.plataforma = "PC";
        this.jugadores = 1;
    }

    public Videojuego(String nombre, String plataforma) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.jugadores = 1;
    }

    public Videojuego(String nombre, String plataforma, int jugadores) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.jugadores = jugadores;
    }

    public void agregarJugadores() {
        jugadores++;
    }

    public void agregarJugadores(int cantidad) {
        jugadores += cantidad;
    }

    public void mostrar() {
        System.out.println(nombre + " (" + plataforma + ") - Jugadores: " + jugadores);
    }

    public static void main(String[] args) {
        Videojuego v1 = new Videojuego("Minecraft", "PC", 5);
        Videojuego v2 = new Videojuego("GTA V", "PlayStation");

        v1.agregarJugadores();
        v2.agregarJugadores(3);

        v1.mostrar();
        v2.mostrar();
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_SEGA_LEONEL_ISAIAS/Polimorfisom0/2.py
class Videojuego:
    def __init__(self, nombre="", plataforma="", jugadores=1):
        self.nombre = nombre
        self.plataforma = plataforma
        self.jugadores = jugadores

    # Sobrecarga simulada de constructor (Python no permite múltiples constructores)
    @classmethod
    def crear_con_datos(cls, nombre, plataforma):
        return cls(nombre, plataforma, 1)

    def agregar_jugadores(self, cantidad=None):
        if cantidad is None:
            self.jugadores += 1
        else:
            self.jugadores += cantidad

    def mostrar(self):
        print(f"{self.nombre} ({self.plataforma}) - Jugadores: {self.jugadores}")

# Ejemplo
v1 = Videojuego("Minecraft", "PC", 5)
v2 = Videojuego.crear_con_datos("GTA V", "PlayStation")
v1.agregar_jugadores()
v2.agregar_jugadores(3)
v1.mostrar()
v2.mostrar()
